alvr/dashboard/src/dashboard/components/settings_controls/presets/higher_order_choice.rs

```diff
@@ -92,6 +93,15 @@ impl Control {
     }
 
     pub fn update_session_settings(&mut self, session_setting_json: &json::Value) {
+        fn values_equal(a: &serde_json::Value, b: &serde_json::Value) -> bool {
+            if let (serde_json::Value::Number(n1), serde_json::Value::Number(n2)) = (a, b) {
+                if let (Some(f1), Some(f2)) = (n1.as_f64(), n2.as_f64()) {
+                    return (f1 - f2).abs() < 1e-6;
```

Comment:
Instead of using an arbitrary constant, you could try using `f64::MIN`. Actually, since these values might be stored in 32 bit, and we usually don't need more than 32 bit, using `f32::MIN` might be safer. You should also fix this for the normal settings, in number.rs, where we check for reset button activation with `!=` between floats which is wrong as here before.

Reply:
f32::MIN is negative value
i think you're referring to epsilon, but f64::EPSILON is too small, f32::EPSILON requires converting from f64 to f32 in comparison
isn't it better to leave as just 1e-6?

Reply:
No, 1e-6 is too big IMO, might break some settings. f32::EPSILONshould be right about what we need to fix the bug and not be any bigger

Reply:
and the reason to use the f32 version of EPSILON is because the actual settings might be 32 bit, and so the server might introduce rounding errors

Reply:
FYI: epsilon for f32 is like, 1e-7, and yeah it means everything gets converted to f32 inside comparison from f64

Reply:
This is everything but how you should do the comparison btw, the errors between floats aren't equal all over the number space (i.e. if you're already comparing very small numbers vs comparing very large ones). We should really use an actual crate for this (or take a known sane algorithm and implement it).

Reply:
i was thinking about using separate crate tbh, yeah

Reply:
Ok so there is crate, it's a bit better but you still have to use epsilon for precision